### DISCUSSION TITLE
WinUI2 vs WinUI3 source code

### DISCUSSION THREAD

**Discussion by deanchalk** [microsoft/microsoft-ui-xaml#8352] [Q&A]
I wanted to familiarise myself with the WinUI 3 C++ source code, but it's not yet available to view.
Would it be reasonable to assume that the WinUI 3 C++ controls source code is very similar to the WinUI2 C++ source code?
If so, it would be a reasonable thing for me to invest time working through the WinUI2 code to better understand WinUI3 controls - is that a good strategy?

Many thanks

**Comment by thatsofia** [comment]
I suggest that you don't invest any time on native Windows development. The future of it is grim. I've gotten burned before.
